### PR TITLE
parser: parse noctal number literals

### PR DESCRIPTION
**Pull request by belochub**
Refs: https://github.com/metarhia/jstp/issues/175
Refs: https://github.com/metarhia/jstp/issues/152

### CODE REVIEW

File: src/jsrs_parser.cc [hunk 1]
```diff
@@ -426,6 +425,55 @@ Local<Value> ParseBigIntegerNumber(Isolate*    isolate,
...
+      break;
+    }
+
+    current_digit = begin[i] - '0';
+    if (current_digit > 7) {
+      is_octal = false;
+    }
+
+    if (!int32_overflow) {
+      result *= 10;
```
Comment (nechaido) [inline src/jsrs_parser.cc]: Shouldn't it be 8?

File: src/jsrs_parser.cc [hunk 2]
```diff
@@ -426,6 +425,55 @@ Local<Value> ParseBigIntegerNumber(Isolate*    isolate,
...
+      result += current_digit;
+    } else {
+      overflow_result *= 10.0;
```
Comment (nechaido) [inline src/jsrs_parser.cc]: Ditto.

File: src/jsrs_parser.cc (hunk 1 above)
Comment (belochub) [inline src/jsrs_parser.cc]: No, because then it will be octal number parsing.

Comment (nechaido) [inline src/jsrs_parser.cc]: Oh, yep. Ok then.

**Comment by aqrln** [comment]
Landed in 726c8d8f7adad04170722f3396a21e7162920f39.

**Pull request closed without merging**